src/passes/Print.cpp

```diff
@@ -835,73 +813,94 @@ struct PrintSExpression : public Visitor<PrintSExpression> {
     printName(curr->value, o) << "))";
   }
   void visitGlobal(Global* curr) {
+    doIndent(o, indent);
     o << '(';
-    printMedium(o, "global ");
-    printName(curr->name, o) << ' ';
-    if (curr->mutable_) {
-      o << "(mut " << printType(curr->type) << ") ";
+    if (curr->imported()) {
+      printMedium(o, "import ");
+      printText(o, curr->module.str) << ' ';
+      printText(o, curr->base.str) << ' ';
+      o << "(global " << curr->name << ' ' << printType(curr->type) << ")";
     } else {
-      o << printType(curr->type) << ' ';
+      printMedium(o, "global ");
+      printName(curr->name, o) << ' ';
+      if (curr->mutable_) {
+        o << "(mut " << printType(curr->type) << ") ";
+      } else {
+        o << printType(curr->type) << ' ';
+      }
+      visit(curr->init);
     }
-    visit(curr->init);
     o << ')';
+    o << maybeNewLine;
   }
   void visitFunction(Function* curr) {
+    doIndent(o, indent);
     currFunction = curr;
     lastPrintedLocation = { 0, 0, 0 };
     o << '(';
-    printMajor(o, "func ");
-    printName(curr->name, o);
-    if (currModule && !minify) {
-      // emit the function index in a comment
-      if (functionIndexes.empty()) {
-        ModuleUtils::BinaryIndexes indexes(*currModule);
-        functionIndexes = std::move(indexes.functionIndexes);
+    if (!curr->imported()) {
```

Comment:
Can we flip this conditional, `if (curr->imported()) { /* import printing logic */ } else { //...`?
The else here is really far from the if, and makes it harder to see the else's precondition. The many intermediate ifs ands nests don't help either.
I'm tempted to say flip to `if (curr->imported())`, copy the `o << maybeNewline;`, and add an early-return, just to avoid more nesting. That may be more controversial, however.

Reply:
Alternatively: extract the imported case to a helper function, `void printImportFunction(Function* curr);`, and handle it totally separately (basically rename the existing visitImport?).
Or: just because Imports aren't separate in the IR, doesn't mean we can't have separate functions to handle them.

Could do something like:
```c++
void printImportCommon(Importable* curr) {
  printMedium(o, "import ");
  printText(o, curr->module.str) << ' ';
  printText(o, curr->base.str) << ' ';
}

void printImportedFunction(Function* curr) {
  printImportCommon(curr);
  if (curr->type.is()) {
    visitFunctionType(currModule->getFunctionType(curr->type), &curr->name);
  }
  o << maybeNewline;
}
```
and reuse for globals and the like.
Probably could keep most of these inline.

Reply:
I like the idea to have separate functions for imported/defined functions/globals, refactoring.